src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,346 @@
+
+    protected ParameterService parameterService;
+    protected ChartService chartService;
+    protected AccountService accountService;
+    protected SubAccountService subAccountService;
+    protected ObjectCodeService objectCodeService;
+    protected SubObjectCodeService subObjectCodeService;
+    protected ProjectCodeService projectCodeService;
+    protected ConfigurationService configurationService;
+
+    @Override
+    public AccountingXmlDocumentAccountingLine createAccountingXmlDocumentAccountingLine(GroupLevel costCenterGroupLevel,
+                                                                                         DefaultKfsAccountForAws defaultKfsAccountForAws) throws IllegalArgumentException {
+        LOG.debug("createAccountingXmlDocumentAccountingLine for " + costCenterGroupLevel.getGroupName());
+        if (!costCenterGroupLevel.isCostCenterGroupLevel()) {
+            throw new IllegalArgumentException(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_INVALID_GROUP_LEVEL_TYPE));
+        }
+
+        String costCenterGroupValue = costCenterGroupLevel.getGroupValue();
+        AccountingXmlDocumentAccountingLine xmlAccountingLine = getEmptyAccountingXmlDocumentAccountingLine();
+        if (StringUtils.countMatches(costCenterGroupValue, "*") == 6) {
+            xmlAccountingLine = parseCostCenterGroupLevel(costCenterGroupLevel.getGroupValue());
+        }
+        else if (StringUtils.countMatches(costCenterGroupValue, "-") == 1) {
+            xmlAccountingLine.setAccountNumber(StringUtils.split(costCenterGroupValue, "-")[0]);
+            xmlAccountingLine.setSubAccountNumber(StringUtils.split(costCenterGroupValue, "-")[1]);
+        }
+        else {
+            xmlAccountingLine.setAccountNumber(costCenterGroupValue);
+        }
+
+        xmlAccountingLine = fixAccountFieldReferences(xmlAccountingLine, defaultKfsAccountForAws.getKfsDefaultAccount());
+        xmlAccountingLine.setAmount(costCenterGroupLevel.getCost());
+        return xmlAccountingLine;
+    }
+
+    private AccountingXmlDocumentAccountingLine fixAccountFieldReferences(AccountingXmlDocumentAccountingLine xmlAccountingLine, String defaultAccountString) {
+        if (!validateChartCode(xmlAccountingLine.getChartCode())) {
+            String defaultChartCode = getDefaultChartCodeFromParameter();
+            xmlAccountingLine.setChartCode(defaultChartCode);
+        }
+
+        if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.ACCOUNT_NONE, xmlAccountingLine.getAccountNumber())) {
+            return getDefaultAccountingLine(defaultAccountString);
+        }
+
+        if (!StringUtils.equalsIgnoreCase(defaultAccountString, CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION) &&
+                !validateAccount(xmlAccountingLine.getChartCode(), xmlAccountingLine.getAccountNumber())) {
+            return getDefaultAccountingLine(defaultAccountString);
```

Comment:
Do we want to return the accounting line here and above?  I think we want to just set the accountNumber, and allow the rest of the validation to continue